File: src/my_project_typescript/easy_problems_typescript/from51To100/signOfProductArray.ts
function arraySign(nums: number[]): number {

    let answer: number = 0
    let count: number = 1
    for (let num of nums){
        count = count * num
    }

    if (count === 0){
        return  0
    }
    if (count > 0){
        return  1
    }
    if (count < 0){
        return -1
    }

    return answer 

};


console.log(arraySign([-1,-2,-3,-4,3,2,1]))

console.log(arraySign([1,5,0,2,-3]))

console.log(arraySign([-1,1,-1,1,-1]))

File: src/my_project/easy_problems/from151To200/sign_of_product_array.py
from typing import List 

class Solution:
    def arraySign(self, nums: List[int]) -> int:

        count: int = 1
        

        for number in nums: 
            count *= number

        if count == 0:
            return 0

        if count > 1:
            return 1

        if count < 1:
            return -1

    

        


        


solution = Solution()

print(solution.arraySign(nums = [-1,-2,-3,-4,3,2,1]))

print(solution.arraySign(nums = [1,5,0,2,-3]))

print(solution.arraySign(nums = [-1,1,-1,1,-1]))
